src/utstring.h

```diff
@@ -121,6 +121,15 @@ do {                                                             \
 
 #define utstring_body(s) ((s)->d)
 
+static char* utstring_release(UT_string *s) {
```

Comment:
Surely the opposite of `utstring_disown` would be `utstring_adopt`!

The longer I look at the existing utstring interface, the more I think it'll never quite line up with my C++ish expectations. Seems like I get a utstring by calling `utstring_new(p)`, but then to put a string into it I have to use `utstring_bincpy(p, "mydata", strlen("mydata"))` (there's no `utstring_strcpy`), and then when I'm done I do `utstring_free(p); p = NULL;` (if I forget that reset-to-null operation, I'll get a double-free if I `utstring_renew(p)` — I think `utstring_renew` is an unusual wart in the API).

With this patch, if I want to "release/disown" the string, I do either `char *s = utstring_body(p); utstring_init(p);` or else `char *s = utstring_release(p); p = NULL;` (again forgetting the reset-to-null might lead to double-frees) — and if I want to "release/disown" a non-alloc'ed utstring, I have to use the first method, because `utstring_release` always wants to `free` the utstring I give it.

I'd want to separate the act of "disowning the buffer" and the act of "destroying the (possibly non-alloc'ed) utstring object itself":

```
_UNUSED_ static char *utstring_disown(UT_string *s) {
  char *ret = s->d;
  s->n = 0; s->i = 0; s->d = NULL;
  return ret;
}
```